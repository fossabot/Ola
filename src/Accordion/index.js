import React from 'react'
import {default as PT} from 'prop-types'
import cx from 'classnames'

import Icon from '../Icon'

const Accordion = ({ children, title, extraClass, ...props }) => {
  const styles = cx('ola_accordion', extraClass)
  return (
    <details className={styles} {...props}>
      <summary className="ola_accordion-header">
        <div className="ola_accordion-header-content">
          <span className="ola_accordion-marker"><Icon name="chevronRight" /></span>
          <span className="ola_accordion-title">{ title }</span>
        </div>
      </summary>
      <div className="ola_accordion-content">
        { children }
      </div>
    </details>
  )
}

Accordion.defaultProps = {
  extraClass: null
}

Accordion.propTypes = {
  /** Title */
  title: PT.oneOfType([
    PT.string,
    PT.arrayOf(PT.node),
    PT.node
  ]).isRequired,
  /** Extra className */
  extraClass: PT.string,
  /** Childen nodes */
  children: PT.oneOfType([
    PT.string,
    PT.arrayOf(PT.node),
    PT.node
  ]).isRequired,
}

export default Accordion
