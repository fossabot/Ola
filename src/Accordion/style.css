.ola_accordion {
    &[open] {
        & .ola_accordion-marker {
            transform: rotate(90deg);
        }
        & .ola_accordion-title {
            color: var(--black);
            text-decoration: none;
        }
    }
}
.ola_accordion-title {
    display: inline-flex;
    align-items: center;
    cursor: default;
    font: var(--font-callout);
    font-weight: var(--bold);
    color: var(--brand);
    text-decoration: underline;
    height: var(--size-8);
    padding-right: var(--size-7);
    outline: 0;
    border-radius: var(--radius-small);

    &:focus {
        box-shadow: 0 0 0 4px var(--brand-focus);
    }
    & * + * {
        margin-left: 10px;
    }
}
.ola_accordion-marker {
    display: block;
    flex: 0 0 auto;
    margin-right: var(--size-3);
    color: var(--gray-light);
    border-radius: 50%;
    transition: transform var(--duration);

    & svg {
        display: block;
    }
}