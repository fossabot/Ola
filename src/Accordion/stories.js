import React from 'react'

import create from '../create'
import Accordion from './'
import Tag from '../Tag'

create('Accordion')
  .add('Default', () => (
    <figure>
      <Accordion title="See more">
        <p>Lorem ipsum dolor sit amet</p>
      </Accordion>
    </figure>
  ))
  .add('Counter', () => (
    <figure>
      <Accordion title={
        <>
            This is the title
            <Tag>23</Tag>
        </>
      }>
        <p>Lorem ipsum dolor sit amet</p>
      </Accordion>
    </figure>
  ))
  .add('Default open', () => (
    <figure>
      <Accordion title="Opened" open={true}>
        <p>Lorem ipsum dolor sit amet</p>
      </Accordion>
    </figure>
  ))
  