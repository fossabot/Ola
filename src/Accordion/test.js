import React from 'react'
import Accordion from './'
import renderer from 'react-test-renderer'

it('Default accordion', () => {
  const tree = renderer
    .create(<Accordion title="See more">Lorem ipsum dolor sit amet</Accordion>)
    .toJSON()

  expect(tree).toMatchSnapshot()
})

it('Open accordion', () => {
  const tree = renderer
    .create(<Accordion title="See more" open>Lorem ipsum dolor sit amet</Accordion>)
    .toJSON()

  expect(tree).toMatchSnapshot()
})

it('Accordion with extra class', () => {
  const tree = renderer
    .create(<Accordion title="See more" className="foo">Lorem ipsum dolor sit amet</Accordion>)
    .toJSON()

  expect(tree).toMatchSnapshot()
})
