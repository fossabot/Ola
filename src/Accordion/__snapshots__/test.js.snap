// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Accordion with extra class 1`] = `
<details
  className="ola_accordion foo"
>
  <summary
    className="ola_accordion-header"
  >
    <div
      className="ola_accordion-header-content"
    >
      <span
        className="ola_accordion-marker"
      >
        <svg
          className="ola_icon is-medium"
          fillRule="evenodd"
          height="28"
          viewBox="0 0 28 28"
          width="28"
        >
          <path
            d="M14 15.586l5.293-5.293a1 1 0 0 1 1.414 1.414l-6 6a1 1 0 0 1-1.414 0l-6-6a1 1 0 0 1 1.414-1.414L14 15.586z"
          />
        </svg>
      </span>
      <span
        className="ola_accordion-title"
      >
        See more
      </span>
    </div>
  </summary>
  <div
    className="ola_accordion-content"
  >
    Lorem ipsum dolor sit amet
  </div>
</details>
`;

exports[`Default accordion 1`] = `
<details
  className="ola_accordion"
>
  <summary
    className="ola_accordion-header"
  >
    <div
      className="ola_accordion-header-content"
    >
      <span
        className="ola_accordion-marker"
      >
        <svg
          className="ola_icon is-medium"
          fillRule="evenodd"
          height="28"
          viewBox="0 0 28 28"
          width="28"
        >
          <path
            d="M14 15.586l5.293-5.293a1 1 0 0 1 1.414 1.414l-6 6a1 1 0 0 1-1.414 0l-6-6a1 1 0 0 1 1.414-1.414L14 15.586z"
          />
        </svg>
      </span>
      <span
        className="ola_accordion-title"
      >
        See more
      </span>
    </div>
  </summary>
  <div
    className="ola_accordion-content"
  >
    Lorem ipsum dolor sit amet
  </div>
</details>
`;

exports[`Open accordion 1`] = `
<details
  className="ola_accordion"
  open={true}
>
  <summary
    className="ola_accordion-header"
  >
    <div
      className="ola_accordion-header-content"
    >
      <span
        className="ola_accordion-marker"
      >
        <svg
          className="ola_icon is-medium"
          fillRule="evenodd"
          height="28"
          viewBox="0 0 28 28"
          width="28"
        >
          <path
            d="M14 15.586l5.293-5.293a1 1 0 0 1 1.414 1.414l-6 6a1 1 0 0 1-1.414 0l-6-6a1 1 0 0 1 1.414-1.414L14 15.586z"
          />
        </svg>
      </span>
      <span
        className="ola_accordion-title"
      >
        See more
      </span>
    </div>
  </summary>
  <div
    className="ola_accordion-content"
  >
    Lorem ipsum dolor sit amet
  </div>
</details>
`;
